app/gui/controller/double-representation/src/identifier.rs

```diff
@@ -191,18 +191,18 @@ impl Hash for Identifier {
 /// Generate an identifier name that is not present in the given sequence.
 ///
 /// The name is generated by taking `base` string and appending subsequent integers.
-pub fn generate_name(
-    base: impl AsRef<str>,
-    unavailable: impl IntoIterator<Item = String>,
+pub fn generate_name<'a>(
+    base: &str,
+    unavailable: impl IntoIterator<Item = &'a str>,
 ) -> FallibleResult<Identifier> {
-    let base = base.as_ref();
-    let is_relevant = |name: &String| name.starts_with(base);
-    let unavailable = unavailable.into_iter().filter(is_relevant).collect::<HashSet<_>>();
+    let unavailable = unavailable
+        .into_iter()
+        .filter_map(|name| name.strip_prefix(base).and_then(|suffix| suffix.parse::<usize>().ok()))
+        .collect::<HashSet<_>>();
```

Comment:
This code is less understandable than before; it was obvious we leave "relevant" unavailable names only, now we do some stripping. The var should be named differently.